Tests/MathLib/AutoCheckTools.h

```diff
@@ -67,6 +68,22 @@ struct IntervalTupleGenerator
     }
 };
 
+/// Generator for MxN fixed size eigen matrices with underlying type T.
+template <typename T, std::size_t M, std::size_t N, typename Gen = generator<T>>
+struct randomEigenMatrixGenerator
+{
+    Gen source;
+
+    using result_type = Eigen::Matrix<T, M, N>;
```

Comment:
Not now. If needed I'll add that possibility (to check for storage order independence of algorithms, for example).